src/input_arguments_handling.jl

```diff
@@ -0,0 +1,122 @@
+# These are the user-provided arguments to `parse_file` or `setup_parser`.
+# We'll use them to construct the `ParsingContext` and `ChunkingContext`,
+# after we sniff the beginning the file to determine the newline,
+# header, etc.
+struct InputArguments
+    header::Union{Nothing,Vector{Symbol}}
+    header_at::Int
+    data_at::Int
+    validate_type_map::Bool
+    default_colname_prefix::String
+    no_quoted_newlines::Bool
+    newlinechar::Union{Nothing,UInt8}
+    deduplicate_names::Bool
+    function InputArguments(header::Vector{Symbol}, data_at, validate_type_map, default_colname_prefix, no_quoted_newlines, newlinechar, deduplicate_names)
+        new(header,  0,           Int(data_at), validate_type_map, default_colname_prefix, no_quoted_newlines, newlinechar, deduplicate_names)
+    end
+
+    function InputArguments(header::Integer,        data_at, validate_type_map, default_colname_prefix, no_quoted_newlines, newlinechar, deduplicate_names)
+        new(nothing, Int(header), Int(data_at), validate_type_map, default_colname_prefix, no_quoted_newlines, newlinechar, deduplicate_names)
+    end
+end
+
+# Fail early if the user is trying to parse something we don't know how to parse
+_is_supported_type(::Type{T}) where {T} = Parsers.supportedtype(T)
+_is_supported_type(::Type{Nothing}) = true # Column skipping
+_is_supported_type(::Type{GuessDateTime}) = true # Our custom DateTime parser that handles multiple subsets of ISO8601
+function _is_supported_type(::Type{FixedDecimal{T,f}}) where {T,f}
+    # https://github.com/JuliaMath/FixedPointDecimals.jl/blob/1328b9a372d2285765a7255f154f09ffdd692508/src/FixedPointDecimals.jl#L83-L91
```

Comment:
why do we link this line of code?
please add a sentence to this comment saying whatever we want to say here